platform/admin/ui/src/main/webapp/js/controllers/SessionTimeout.controller.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+/* jshint browser: true */
+/* global define */
+define(['jquery',
+        'underscore',
+        'marionette',
+        'js/wreqr',
+        'js/views/SessionTimeoutModal.view'
+    ], function ($, _, Marionette, wreqr, SessionTimeoutModal) {
+        'use strict';
+        var SessionTimeoutController;
+
+        var _this = null;
+        var idleNoticeDuration = 60000;
+        var idleTimeoutThreshold = 900000;  // Length of inactivity that will trigger user timeout
+                                            // See STIG V-69243
+
+        SessionTimeoutController = Marionette.Object.extend({
+            initialize: function () {
+                _this = this;
+                _this.initTimer();
+
+                wreqr.vent.on("sessionRenewed", _this.initTimer);
+            },
+
+            initTimer: function () {
+                var timer = null;
```

Comment:
❗️  It looks like you can end up with a timer that is not cleared if sessionRenewed is called and timer is set to null while active.